drivers/counter/counter_smartbond_timer.c

```diff
@@ -59,16 +59,22 @@ struct counter_smartbond_config {
 	LOG_INSTANCE_PTR_DECLARE(log);
 };
 
-#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
+#if defined(CONFIG_PM_DEVICE)
 static void counter_smartbond_pm_policy_state_lock_get(struct counter_smartbond_data *data)
 {
 	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
 		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
 	}
+#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	pm_device_runtime_get(dev);
+#endif
```

Comment:
These `ifdef`s are not necessary, because the functions are already declared as empty static inline functions when PM_DEVICE_RUNTIME is not enabled.
```suggestion
	pm_device_runtime_get(dev);
```

```
static inline int pm_device_runtime_get(const struct device *dev)
{
	ARG_UNUSED(dev);
	return 0;
}
```

Reply:
Removed unnecessary `ifdef`.